notebook/static/tree/js/notebooklist.js

```diff
@@ -697,6 +707,71 @@ define([
         });
     };
 
+    NotebookList.prototype.move_selected = function() {
+        var that = this;
+
+        // TODO(nhdaly): Support moving multiple items at once.
+        if (that.selected.length !== 1){
+            return;
+        }
+
+        var item_path = that.selected[0].path;
+        var item_name = that.selected[0].name;
+        var item_type = that.selected[0].type;
+
+        // Open a dialog to enter the new path, with current path as default.
+        var input = $('<input/>').attr('type','text').attr('size','25').addClass('form-control')
+            .val(utils.url_path_join('/', that.notebook_path));
+        var dialog_body = $('<div/>').append(
+            $("<p/>").addClass("rename-message")
+                .text('Enter new destination directory path for '+ item_type + ':')
+        ).append(
+            $("<br/>")
+        ).append(input);
+        var d = dialog.modal({
+            title : "Move "+ item_type,
+            body : dialog_body,
+            buttons : {
+                Cancel : {},
+                OK : {
+                    class: "btn-primary",
+                    click: function() {
+                      // Construct the new path using the user input and its name.
+                        var new_path = utils.url_path_join(input.val(), item_name)
+                        that.contents.rename(item_path, new_path).then(function() {
+                            that.load_list();
+                        }).catch(function(e) { 
+                            dialog.modal({
+                                title: "Move Failed",
+                                body: $('<div/>')
+                                    .text("An error occurred while moving \"" + item_name + "\" from \"" + item_path + "\" to \"" + new_path + "\".")
+                                    .append($('<div/>')
+                                        .addClass('alert alert-danger')
+                                        .text(e.message || e)),
+                                buttons: {
+                                    OK: {'class': 'btn-primary'}
+                                }
+                            });
+                            console.warn('Error durring moving :', e);
+                        });
+                    }
+                }
+            },
+            open : function () {
+                // Upon ENTER, click the OK button.
+                input.keydown(function (event) {
+                    if (event.which === keyboard.keycodes.enter) {
+                        d.find('.btn-primary').first().click();
+                        return false;
+                    }
+                });
+                input.focus();
+                // Highlight the current path in the input box.
+                input.select();
```

Comment:
@ellisonbg + @captainsafia: Also, I wonder about how well the github style interface would support moving multiple files? I have opened #942 to support moving multiple files from the Move dialogue.

I guess maybe you could do the same thing as the Github UI, but leave off the file name?